clothcalc_placeholder.js

```diff
@@ -3952,69 +3974,106 @@
-                            else { createMenu(e); }; });
+                            if (w.Character.homeTown.town_id !== 0 && forts.length == 0) sleepHome();
+                            else prepareMenu(e); });
                 };
 
-                var createMenu = function(e) {
-                    if (forts.length == 0) { return; };
-                    var pos = Map.getLastPosition();
-                    pos.x = pos[0];
-                    pos.y = pos[1];
-                    for (var i=0; i < forts.length; i++) {
-                        forts[i].distance = w.Map.calcWayTime(pos,forts[i]);
-                    };
-                    forts.sort(function(a, b){ return(a.distance==b.distance)?0:(a.distance>b.distance)?1:-1;});
+                var createMenu = function(arr, e, length) {
                     var selectbox = (new west.gui.Selectbox(true))
-                        .addListener(function (key) { switch (key) { case 0: sleepHotel(); break;
-                                                                    default: sleepFort(key); break; }; })
-                        .addItem(0, '#HOTEL#&nbsp;' + w.Map.calcWayTime(pos, w.Character.homeTown).formatDuration());
-                    for (var i=0; i < forts.length; i++) {
-                        if (forts[i].stage !== 0) {
-                            selectbox.addItem(forts[i].id, '#STAGE#&nbsp;' + forts[i].stage + '&nbsp;' + forts[i].distance.formatDuration() + '&nbsp;|&nbsp;' + forts[i].name);
-                        };
-                    };
+                        .addListener(function (key) { switch (key) {
+                            case 'home': sleepHome(); break;
+                            default: sleep(key); break; }
+                        });
+                    if (w.Character.homeTown.town_id !== 0)
+                        selectbox.addItem('home', '#HOTEL#&nbsp;' + w.Map.calcWayTime(pos, w.Character.homeTown).formatDuration());
+                    for (var i=0; i < length; i++)
+                        if (arr[i].stage !== 0)
+                            selectbox.addItem(i, '#STAGE#&nbsp;' + arr[i].stage + '&nbsp;' + arr[i].distance.formatDuration() + '&nbsp;|&nbsp;' + arr[i].name);
                     $(selectbox.elContent).addClass("twdb_sleepmenu");
                     selectbox.show(e);
+                    loading = false;
+                };
+                
+                var nearTowns = function(i, max, e) {
+                    if (i == max)
+                        createMenu(towns, e, max);
+                    else if (towns[i].hasOwnProperty('stage'))
+                        nearTowns(++i, max, e);
+                    else
+                        Ajax.remoteCallMode('building_hotel', 'get_data', {town_id: towns[i].town_id}, function (data){
+                            if (data.error)
+                                return new UserMessage(data.msg).show();
```

Comment:
Yes, when you get in a new town, you will just sleep at home, forts won't be loaded.
When you leave the town, it will just display an empty selectbox.